File: src/api/models/Message.js
import mongoose from "mongoose";

const schema = mongoose.Schema;

const userSchema = new schema(
  {
    content: {
      type: String,
      required: false,
      default: "",
    },
    contentUrl: {
      type: String,
      required: false,
      default: "",
    },
    duration: {
      type: Number,
      required: false,
      default: 0,
    },
    type: {
      type: Number,
      required: true,
    },
    deleteBy: [
      {
        type: schema.Types.ObjectId,
        ref: "users",
      },
    ],
    seenBy: [
      {
        type: schema.Types.ObjectId,
        ref: "users",
      },
    ],
    sendBy: {
      type: schema.Types.ObjectId,
      ref: "users",
    },
    conversation: {
      type: schema.Types.ObjectId,
      ref: "conversations",
    },
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now },
  },
  { strict: false }
);

const Message = mongoose.model("messages", userSchema);

export default Message;

File: app/src/main/java/com/example/chattomate/models/Message.java
package com.example.chattomate.models;

import java.util.ArrayList;

public class Message {
    public String conversation;
    public String _id;
    public String content;
    public String contentUrl;
    public String sendAt;
    public ArrayList<Friend> seenBy;
    public Friend sendBy;
    public boolean delete;
    public String type;

    public Message(String content) {
        this.content = content;
    }

    public Message(String conversation, String _id, String content, String contentUrl, String sendAt, ArrayList<Friend> seenBy, Friend sendBy, boolean delete, String type) {
        this.conversation = conversation;
        this._id = _id;
        this.content = content;
        this.contentUrl = contentUrl;
        this.sendAt = sendAt;
        this.seenBy = seenBy;
        this.sendBy = sendBy;
        this.delete = delete;
        this.type = type;
    }

    public Message() {}
}
